apps/src/javalab/JavalabEditorTabMenu.jsx

```diff
@@ -32,102 +32,68 @@ class JavalabTabMenuComponent extends Component {
-          type="button"
-          key="rename"
-          onClick={renameFromTabMenu}
-          style={styles.anchor}
+    let elements = [
+      <a onClick={renameFromTabMenu} key="rename">
+        {javalabMsg.rename()}
+      </a>,
+      <a onClick={deleteFromTabMenu} key="delete">
+        {javalabMsg.delete()}
+      </a>
+    ];
+    if (showVisibilityOption && !fileIsVisible) {
+      elements.push(
+        <a
+          key="starter"
+          onClick={() => {
+            changeFileTypeFromTabMenu(
+              true /*isVisible*/,
+              false /*isValidation*/
+            );
+          }}
         >
-          Rename
-        </button>
-        <button
-          type="button"
-          key="delete"
-          onClick={deleteFromTabMenu}
-          style={styles.anchor}
+          {javalabMsg.makeStarter()}
+        </a>
+      );
+    }
+    if (showVisibilityOption && (fileIsVisible || fileIsValidation)) {
+      elements.push(
+        <a
+          key="support"
+          onClick={() => {
+            changeFileTypeFromTabMenu(
+              false /*isVisible*/,
+              false /*isValidation*/
+            );
+          }}
         >
-          Delete
-        </button>
-        {showVisibilityOption && !fileIsVisible && (
-          <button
-            type="button"
-            key="starter"
-            onClick={() => {
-              changeFileTypeFromTabMenu(
-                true /*isVisible*/,
-                false /*isValidation*/
-              );
-            }}
-            style={styles.anchor}
-          >
-            Make starter file
-          </button>
-        )}
-        {showVisibilityOption && (fileIsVisible || fileIsValidation) && (
-          <button
-            type="button"
-            key="support"
-            onClick={() => {
-              changeFileTypeFromTabMenu(
-                false /*isVisible*/,
-                false /*isValidation*/
-              );
-            }}
-            style={styles.anchor}
-          >
-            Make support file
-          </button>
-        )}
-        {showVisibilityOption && !fileIsValidation && (
-          <button
-            type="button"
-            key="validation"
-            onClick={() => {
-              changeFileTypeFromTabMenu(
-                false /*isVisible*/,
-                true /*isValidation*/
-              );
-            }}
-            style={styles.anchor}
-          >
-            Make validation file
-          </button>
-        )}
-        <button
-          type="button"
-          key="cancel"
-          onClick={cancelTabMenu}
-          style={styles.anchor}
+          {javalabMsg.makeSupport()}
+        </a>
+      );
+    }
+    if (showVisibilityOption && !fileIsValidation) {
+      elements.push(
+        <a
+          key="validation"
+          onClick={() => {
+            changeFileTypeFromTabMenu(
+              false /*isVisible*/,
+              true /*isValidation*/
```

Comment:
(May be out of scope for this change) Can the branching conditions be refactored a bit here? i.e. I think having changeFileTypeFromTabMenu take two inputs is complicating things here. Or perhaps one `if (showVisibilityOption)` followed by `if (!fileIsValidation)` etc.

Reply:
I refactored the if statements, updating the function is out of scope for this work item.